app/src/main/java/com/tari/android/wallet/service/BootDeviceReceiver.kt

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2020 The Tari Project
+ *
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the
+ * following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of
+ * its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ * CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.tari.android.wallet.service
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import com.orhanobut.logger.Logger
+import com.tari.android.wallet.application.TariWalletApplication
+import com.tari.android.wallet.di.WalletModule
+import com.tari.android.wallet.util.SharedPrefsWrapper
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Named
+
+/**
+ * This receiver is responsible for starting the service after boot finish.
+ *
+ * @author The Tari Development Team
+ */
+class BootDeviceReceiver : BroadcastReceiver() {
+
+    @Inject
+    @Named(WalletModule.FieldName.walletFilesDirPath)
+    lateinit var walletFilesDirPath: String
+    @Inject
+    internal lateinit var sharedPrefsWrapper: SharedPrefsWrapper
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (context == null || intent == null) return
+        Logger.d("Boot device broadcast received.")
+
+        (context.applicationContext as TariWalletApplication).appComponent.inject(this)
+
+        val walletExists = File(walletFilesDirPath).list()!!.isNotEmpty()
+        if (walletExists && sharedPrefsWrapper.onboardingAuthSetupCompleted && Intent.ACTION_BOOT_COMPLETED == intent.action) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
You can use `ContextCompat.startForegroundService`, see `HomeActivity.kt` for an example.